util/flags/manager.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package flags implements the manager options utilities.
+package flags
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apiserver/pkg/server/routes"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/logs"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/filters"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+// ManagerOptions provides command line flags for manager options.
+type ManagerOptions struct {
```

Comment:
all exported fields should have comments even if deprecated.  it's a good practice.
aren't linters going to complain about this?

Reply:
No the linter is not complaining. I think it complains about exported structs and methods, but not about exported fields of exported structs

Reply:
Added some comments. I don't think it would additionally make sense to duplicate the entire flag descriptions on the fields as well.